File: Python/905.SortArrayByParity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        even=0
        odd=len(A)-1
        
        while even<odd:
            if A[even]%2 == 0:
                even+=1
            else:
                if A[odd]%2 == 0:
                    tmp=A[odd]
                    A[odd]=A[even]
                    A[even]=tmp
                odd-=1
        return A
        

File: C++/905. SortArrayByParity.cpp
//Given an array A of non-negative integers, return an array consisting of all the even elements of A, followed by all the odd elements of A.
//Check if odd, erase it from the array and push the element at the back

class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int n=A.size();
        int i=0,count=0;
        while(count<n){
            if(A[i]%2!=0){
                A.push_back(A[i]);
                A.erase(A.begin()+i);
            }
            else
                i++;
            count++;
        }
        return A;
    }
};
